firestore.rules

```diff
@@ -5,13 +5,16 @@ service cloud.firestore {
     	allow read: if true;
       allow write, delete: if userId == request.auth.uid;
       match /{document=**} {
-      	allow read: if true;
-        allow write, delete: if userId == request.auth.uid;
+      	allow read, write, delete: if true;        
```

Comment:
I didn't think ahead on this, but we don't want anyone to be able to write to any other character's document. Thankfully, firestore allows us to set per-field restrictions (see this link https://firebase.google.com/docs/firestore/security/rules-fields#allowing_only_certain_fields_to_be_changed). We might be able to change this to 

```
allow read: if true;
allow write, delete: if userId == request.auth.uid
allow update: if request.resource.data.diff(resource.data).affectedKeys().hasOnly("gmId")
```

I haven't tested this at all, so if it doesn't work let me know and I can play around with it once these changes get merged!

Reply:
I updated the rules to something that should work better.